### PR TITLE
Use random folder under user profile temp folder

### PR DESCRIPTION
**Pull request by 928PJY**
[AB#224063](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/224063)

### CODE REVIEW

File: src/build/buildController.ts [hunk 1]
```diff
@@ -131,11 +132,16 @@ export class BuildController {
 
         try {
             let [localRepositoryUrl, originalRepositoryUrl] = await this.retrieveRepositoryInfo(localRepositoryPath, credential.userInfo.userToken);
+            let outputFolderPath = process.env.VSCODE_DOCS_BUILD_EXTENSION_OUTPUT_FOLDER
+                                     ? process.env.VSCODE_DOCS_BUILD_EXTENSION_OUTPUT_FOLDER
+                                     : getRandomOutputFolder();
```
Comment (mingwli) [inline src/build/buildController.ts]: let outputFolderPath = process.env.VSCODE_DOCS_BUILD_EXTENSION_OUTPUT_FOLDER || getRandomOutputFolder(); #Resolved

File: src/utils/utils.ts [hunk 2]
```diff
@@ -114,4 +116,17 @@ export async function killProcessTree(pid: number, signal?: string | number) {
...
+}
+
+export function getRandomOutputFolder() {
+    let randomFolder = Math.random().toString(36).substring(7);
+    return path.join(tempDirectory, randomFolder);
+}
+
+export function normalizeDriveLetter(filePath: string) {
+    if(process.platform === 'win32'){
+        return filePath.replace(/^([A-Z]):/, (match, driver) => `${driver.toLowerCase()}:`);
```
Comment (live1206) [inline src/utils/utils.ts:128]: why do we need to normalize driver letter?

**Review by live1206 — APPROVED** [review]

File: src/utils/utils.ts (hunk 2 above)
Comment (928PJY) [reply src/utils/utils.ts:128]: the `__dirname` is not stable for driver letter case

---
In reply to: [429754881](https://github.com/docascode/vscode-docs-build/pull/106#discussion_r429754881) [](ancestors = 429754881)

**Review by mingwli — APPROVED** [review]
:shipit:

**Pull request merged**